src/git-repository-async.js

```diff
@@ -0,0 +1,870 @@
+            heads.push(ref.name())
+          }
+        }
+        return {heads, remotes, tags}
+      })
+  }
+
+  // Public: Get the SHA for the given reference.
+  //
+  // * `reference` The {String} reference to get the target of.
+  // * `path` An optional {String} path in the repo to get the reference target
+  //   for. Only needed if the repository contains submodules.
+  //
+  // Returns a {Promise} which resolves to the current {String} SHA for the
+  // given reference.
+  getReferenceTarget (reference, _path) {
+    return this.getRepo(_path)
+      .then(repo => Git.Reference.nameToId(repo, reference))
+      .then(oid => oid.tostrS())
+  }
+
+  // Reading Status
+  // ==============
+
+  // Public: Resolves true if the given path is modified.
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to a {Boolean} that's true if the `path`
+  // is modified.
+  isPathModified (_path) {
+    return this._filterStatusesByPath(_path).then(statuses => {
+      return statuses.filter(status => status.isModified()).length > 0
+    })
+  }
+
+  // Public: Resolves true if the given path is new.
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to a {Boolean} that's true if the `path`
+  // is new.
+  isPathNew (_path) {
+    return this._filterStatusesByPath(_path).then(statuses => {
+      return statuses.filter(status => status.isNew()).length > 0
+    })
+  }
+
+  // Public: Is the given path ignored?
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to a {Boolean} that's true if the `path`
+  // is ignored.
+  isPathIgnored (_path) {
+    return this.repoPromise.then(repo => Git.Ignore.pathIsIgnored(repo, _path))
+  }
+
+  // Get the status of a directory in the repository's working directory.
+  //
+  // * `directoryPath` The {String} path to check.
+  //
+  // Returns a promise resolving to a {Number} representing the status. This value can be passed to
+  // {::isStatusModified} or {::isStatusNew} to get more information.
+  getDirectoryStatus (directoryPath) {
+    let relativePath
+    // XXX _filterSBD already gets repoPromise
+    return this.repoPromise
+      .then(repo => {
+        relativePath = this.relativize(directoryPath, repo.workdir())
+        return this._filterStatusesByDirectory(relativePath)
+      })
+      .then(statuses => {
+        return Promise.all(statuses.map(s => s.statusBit())).then(bits => {
+          let directoryStatus = 0
+          const filteredBits = bits.filter(b => b > 0)
+          if (filteredBits.length > 0) {
+            filteredBits.forEach(bit => directoryStatus |= bit)
+          }
+
+          return directoryStatus
+        })
+      })
+  }
+
+  // Refresh the status bit for the given path.
+  //
+  // Note that if the status of the path has changed, this will emit a
+  // 'did-change-status' event.
+  //
+  // path    :: String
```

Comment:
This appears to be in a different documentation format than the other methods in this file.